misc-utils/lsclocks.c

```diff
@@ -422,6 +437,70 @@ static void add_dynamic_clocks_from_discovery(struct libscols_table *tb,
 	globfree(&state);
 }
 
+static void add_rtc_clock_from_path(struct libscols_table *tb,
+				    const int *columns, size_t ncolumns,
+				    const char *path, bool explicit)
+{
+	int fd, rc;
+	struct rtc_time rtc_time;
+	struct tm tm = { 0 };
+	struct timespec now = { 0 }, resolution = { .tv_nsec = -1 };
+
+	fd = open(path, O_RDONLY);
+	if (fd == -1) {
+		if (explicit)
+			err(EXIT_FAILURE, _("Could not open %s"), path);
+		else
+			return;
+	}
+
+	rc = ioctl(fd, RTC_RD_TIME, &rtc_time);
+	if (rc)
+		err(EXIT_FAILURE,
+		    _("ioctl(RTC_RD_NAME) to %s to read the time failed"), path);
+
+	tm.tm_sec  = rtc_time.tm_sec;
+	tm.tm_min  = rtc_time.tm_min;
+	tm.tm_hour = rtc_time.tm_hour;
+	tm.tm_mday = rtc_time.tm_mday;
+	tm.tm_mon  = rtc_time.tm_mon;
+	tm.tm_year = rtc_time.tm_year;
+	tm.tm_wday = rtc_time.tm_wday;
+	tm.tm_yday = rtc_time.tm_yday;
+
+	now.tv_sec = mktime(&tm);
+
+	struct clockinfo clockinfo = {
+		.type = CT_RTC,
+		.no_id = true,
+		.id_name = path,
+		.name = path,
+	};
+	add_clock_line(tb, columns, ncolumns, &clockinfo, &now, &resolution);
+
+	close(fd);
+}
+
+static void add_rtc_clocks_from_discovery(struct libscols_table *tb,
+					  const int *columns, size_t ncolumns)
+{
+	int rc;
+	size_t i;
+	glob_t state;
+
+	rc = glob("/dev/rtc*", 0, NULL, &state);
+	if (rc == GLOB_NOMATCH)
+		return;
```

Comment:
It also reports symlinks, and the result is duplicated:

```
# ls -la /dev/rtc*
lrwxrwxrwx 1 root root      4 Apr 26 10:15 /dev/rtc -> rtc0
crw------- 1 root root 250, 0 Apr 26 10:15 /dev/rtc0

# lsclocks
...
  /dev/rtc         rtc  1691058519.000000000       2023-08-03T10:28:39.000000000+00:00
  /dev/rtc0        rtc  1691058519.000000000       2023-08-03T10:28:39.000000000+00:00
```
Would it be possible to detect that `rtc0` and `rtc` point to the same device?